server/sonar-server/src/test/java/org/sonar/server/rule/index/RuleIndexTest.java

```diff
@@ -595,115 +570,278 @@ public void search_by_profile_and_active_severity() {
+    ruleIndexer.indexRuleExtension(organization, rule.getKey());
+
+    RuleQuery query = new RuleQuery()
+      .setOrganizationUuid(organization.getUuid());
+    SearchOptions options = new SearchOptions().addFacets(singletonList(FACET_TAGS));
+
+    SearchIdResult result = index.search(query, options);
+    assertThat(result.getFacets().get(FACET_TAGS)).contains(entry("bla", 1L));
+  }
+
+  @Test
+  public void tags_facet_should_return_top_10_items() {
+    // default number of items returned in facets = 10
+    RuleDefinitionDto rule = insertRuleDefinition(setSystemTags("tag1", "tag2", "tag3", "tag4", "tag5", "tag6", "tag7", "tag8", "tag9", "tagA", "tagB"));
+
+    RuleQuery query = new RuleQuery()
+      .setOrganizationUuid(dbTester.getDefaultOrganization().getUuid());
+    SearchOptions options = new SearchOptions().addFacets(singletonList(FACET_TAGS));
+    SearchIdResult result = index.search(query, options);
+    assertThat(result.getFacets().get(FACET_TAGS)).containsExactly(entry("tag1", 1L), entry("tag2", 1L), entry("tag3", 1L), entry("tag4", 1L), entry("tag5", 1L),
+      entry("tag6", 1L), entry("tag7", 1L), entry("tag8", 1L), entry("tag9", 1L), entry("tagA", 1L));
+  }
+
+  @Test
+  public void tags_facet_should_include_matching_selected_items() {
+    // default number of items returned in facets = 10
+    RuleDefinitionDto rule = insertRuleDefinition(setSystemTags("tag1", "tag2", "tag3", "tag4", "tag5", "tag6", "tag7", "tag8", "tag9", "tagA", "tagB"));
+
+    RuleQuery query = new RuleQuery()
+      .setOrganizationUuid(dbTester.getDefaultOrganization().getUuid())
+      .setTags(singletonList("tagB"));
+    SearchOptions options = new SearchOptions().addFacets(singletonList(FACET_TAGS));
+    SearchIdResult result = index.search(query, options);
+    assertThat(result.getFacets().get(FACET_TAGS).entrySet()).extracting(e -> entry(e.getKey(), e.getValue())).containsExactly(
+
+      // check that selected item is added, although there are 10 other items
+      entry("tagB", 1L),
+
+      entry("tag1", 1L), entry("tag2", 1L), entry("tag3", 1L), entry("tag4", 1L), entry("tag5", 1L), entry("tag6", 1L), entry("tag7", 1L), entry("tag8", 1L), entry("tag9", 1L),
+      entry("tagA", 1L));
+  }
+
+  @Test
+  public void tags_facet_should_not_find_tags_of_any_other_organization() {
+    OrganizationDto organization1 = dbTester.organizations().insert();
+    OrganizationDto organization2 = dbTester.organizations().insert();
+
+    RuleDefinitionDto rule = insertRuleDefinition(setSystemTags());
+    dbTester.rules().insertOrUpdateMetadata(rule, organization1, setTags("bla"));
+
+    RuleQuery query = new RuleQuery()
+      .setOrganizationUuid(organization2.getUuid());
+    SearchOptions options = new SearchOptions().addFacets(singletonList(FACET_TAGS));
+
+    SearchIdResult result = index.search(query, options);
+    assertThat(result.getFacets().get(FACET_TAGS)).contains();
```

Comment:
This test succeeds if tag facet feature is broken. I suggest to add a tag on organization2.